src/aiida/transports/transport.py

```diff
@@ -39,15 +43,22 @@ def validate_positive_number(ctx, param, value):
     return value
 
 
-class Transport(abc.ABC):
-    """Abstract class for a generic transport (ssh, local, ...) contains the set of minimal methods."""
+def path_to_str(path: TransportPath) -> str:
+    """Convert an instance of TransportPath = Union[str, Path, PurePosixPath] instance to a string."""
+    # We could check if the path is a Path or PurePosixPath instance, but it's too much overhead.
+    return str(path)
+
+
+class _BaseTransport:
+    """Abstract class for a generic blocking transport.
+    A plugin inhereting from this class should implement the blocking methods, only."""
 
     # This will be used for ``Computer.get_minimum_job_poll_interval``
     DEFAULT_MINIMUM_JOB_POLL_INTERVAL = 10
 
     # This is used as a global default in case subclasses don't redefine this,
     # but this should  be redefined in plugins where appropriate
-    _DEFAULT_SAFE_OPEN_INTERVAL = 30.0
+    _DEFAULT_SAFE_OPEN_INTERVAL = 3.0
```

Comment:
If the class is going to be the base class for the transport classes that will inheritant this one to get the default methods implementation to avoid code duplication. Will these class attributes being redefined (are these default attributes supposed to be redefined?)